src/components/decimal/decimal.component.js

```diff
@@ -227,17 +227,24 @@ const Decimal = ({
     prevControlledRef.current = isControlled;
   }, [isControlled]);
 
+  const prevValue = usePrevious(value);
+  const componentShouldUpdate = stateValue === value;
+
   useEffect(() => {
     const unformattedValue = toStandardDecimal(stateValue);
```

Comment:
IMHO the naming here could be a little clearer, as we discussed on slack something like `standardDecimalValue` might be more intuitive but I'm also okay with you keeping it as is if you disagree